### PR TITLE
Added AnnotatedSlice entity schema

### LINKED ISSUE #79: Add the annotatedSlice schema to the morphology module

### PR DESCRIPTION
**Pull request by srjimenez**
- schema and test files

Fix #79

### CODE REVIEW

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  "imports": [
+    "{{base}}/schemas/bbp/core/entity/v0.1.0",
+    "{{base}}/schemas/bbp/core/quantitativevalue/v0.1.0"
+  ],
+  "shapes": 
+    [
+      {
+        "@id": "this:AnnotatedSliceShape",
+        "@type": "sh:NodeShape",
+        "description": "Annotated slice shape definition. An annotated slice should have an IRI as identifier",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: I would suggest to remove this as we set "nodeKind": "sh:BlankNodeOrIRI".

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          },
+          {
+            "property" : [ {
+              "path" : "nsg:AnnotationAngle",
+              "name" : "Annotation angle",
+              "description" : "Angle at which the slice is as defined during the annotation process",
+              "node": "bbpquantValue:QuantitativeValueShape",
+              "maxCount" : 1
+            }, {
+              "path" : "nsg:AnnotatorComment",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: I would suggest to use rdfs:comment as path value

Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: There are several "comment" fields. Some are for the agents to put a specific info on the activity, in this case the comment is only for the "Annotator" and specific to the "Annotation" activity. It is not a general comment.
In the "reconstruction" I also have a "ReconstructorComment".

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json (hunk 1 above)
Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: OK updated

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json (hunk 2 above)
Comment (MFSY) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: Okay. I still suggest to use a nsg:comment property for now and adopt later the following strategy when modelling comments:
 * if you care about provenance of a given comment (who comment, when, in which context ?) then it should be externalize as an entity with an identifier whom provenance can be asserted.
* sometimes one just want to put a text as a comment: in that case nsg:comment can have psd:string as datatype. 

To completely solve this, a model for storing comments is needed. I've created an [issue](https://github.com/BlueBrain/nexus-bbp-domains/issues/95) for that.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+              {
+                "path": "dcterms:hasPart",
+                "name": "Patched cell collection",
+                "description": "The collections of labeled cells",
+                "class": "nsg:LabeledCellCollection",
+                "minCount": 1
+              },
+              {
+                "path": "prov:wasRevisionOf",
+                "name": "fixedStainedSlice",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: I would suggest to put "Fixed stained slice" as name.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                "description": "The fixedStainedSlice this annotatedSlice is a revision of",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: I would suggest to use plain sentence in descriptions like:
"The fixed stained slice this annotated slice is a revision of."

Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: As discussed we agreed on the following description:
    "description": "The FixedStainedSlice this AnnotatedSlice is a revision of.",

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json (hunk 3 above)
Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: ok

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json (hunk 2 above)
Comment (MFSY) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/annotatedslice/v0.1.0.json]: Let keep "nsg:AnnotatorComment" until this [issue](https://github.com/BlueBrain/nexus-bbp-domains/issues/95) is solved.

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/annotatedslice/v0.1.0/wrong-type.json [hunk 5]
```diff
@@ -22,7 +22,7 @@
     }
   ],
   "wasRevisionOf": {
-    "@id": "abdcd005-e302-49c4-9874-18d1ff11c141",
+    "@id": "{{base}}/data/bbp/experiment/fixedstainedslice/v0.1.0/abdcd005-e302-49c4-9874-18d1ff11c141",
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/annotatedslice/v0.1.0/wrong-type.json]: Shouldn't "fixedstainedslice" part of  bbp/morphology ?

File: modules/bbp-morphology/src/test/resources/data/bbp/morphology/annotatedslice/v0.1.0/min-fields.json [hunk 6]
```diff
@@ -19,12 +19,12 @@
...
   "hasPart": [
     {
-      "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/experiment/labeledcellcollection/v0.1.2/ab880634-3bf5-4872-a34c-b6bbbca8c0a4",
+      "@id": "{{base}}/data/bbp/experiment/labeledcellcollection/v0.1.2/ab880634-3bf5-4872-a34c-b6bbbca8c0a4",
       "@type": "nsg:LabeledCellCollection"
     }
   ],
   "wasRevisionOf": {
-    "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/experiment/fixedstainedslice/v0.1.0/abdcd005-e302-49c4-9874-18d1ff11c141",
+    "@id": "{{base}}/data/bbp/experiment/fixedstainedslice/v0.1.0/abdcd005-e302-49c4-9874-18d1ff11c141",
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/data/bbp/morphology/annotatedslice/v0.1.0/min-fields.json]: Shouldn't this be part of bbp/morphology ?

**Pull request merged**